### PR TITLE
buildkite-agent service: declarative hooks and extraConfig option

### PR DESCRIPTION
**Pull request by rvl**
###### Motivation for this change

I thought it would be better to be able to set hook scripts as module options rather than manually creating a directory with script files.

There are other possible settings in the buildkite-agent config file which aren't covered by module options, so I have added an `extraConfig` option to handle those.

There are also other minor tweaks to the module.

The option changes are backwards compatible.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested NixOS module with Buildkite service.
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

/cc @deepfire @zimbatm @domenkozar

### CODE REVIEW

File: nixos/modules/services/continuous-integration/buildkite-agent.nix [hunk 1]
```diff
@@ -101,7 +197,7 @@ in
 
     systemd.services.buildkite-agent =
       let copy = x: target: perms:
-                 "cp -f ${x} ${target}; ${pkgs.coreutils}/bin/chmod ${toString perms} ${target}; ";
+                 "${pkgs.coreutils}/bin/cp -f ${x} ${target}; ${pkgs.coreutils}/bin/chmod ${toString perms} ${target}; ";
```
Comment (domenkozar) [inline nixos/modules/services/continuous-integration/buildkite-agent.nix]: (nitpick) Maybe we could just add coreutils to `path = cfg.runtimePackages ++ [ pkgs.coreutils]` and we don't need to hardcode them here (for readability)?

**Review by domenkozar — APPROVED** [review]
With one nitpick, looking good! :)

File: nixos/modules/services/continuous-integration/buildkite-agent.nix (hunk 1 above)
Comment (rvl) [reply nixos/modules/services/continuous-integration/buildkite-agent.nix]: Good idea, thanks. I have fixed that.

**Comment by rvl** [comment]
Thanks @zimbatm @domenkozar :-)

**Pull request merged**